File: Task/Events/C-sharp/events.cs
using System;
using System.Timers;

class Program
{
    static void Main()
    {
        var timer = new Timer(1000);
        timer.Elapsed += new ElapsedEventHandler(OnElapsed);
        Console.WriteLine(DateTime.Now);
        timer.Start();
        Console.ReadLine();
    }

    static void OnElapsed(object sender, ElapsedEventArgs eventArgs)
    {
        Console.WriteLine(eventArgs.SignalTime);
        ((Timer)sender).Stop();
    }
}

File: Task/Events/F-Sharp/events.fs
open System
open System.Timers

let onElapsed (sender : obj) (eventArgs : ElapsedEventArgs) =
    printfn "%A" eventArgs.SignalTime
    (sender :?> Timer).Stop()

[<EntryPoint>]
let main argv =
    let timer = new Timer(1000.)
    timer.Elapsed.AddHandler(new ElapsedEventHandler(onElapsed))
    printfn "%A" DateTime.Now
    timer.Start()
    ignore <| Console.ReadLine()
    0
